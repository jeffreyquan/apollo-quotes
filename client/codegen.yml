schema: http://localhost:5000/graphql
generates:
  ./types.ts:
    plugins:
      - typescript
