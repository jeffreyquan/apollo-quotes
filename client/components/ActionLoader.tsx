import React from "react";
import ClipLoader from "react-spinners/ClipLoader";
import { ThemeConsumer } from "styled-components";

export const ActionLoader: React.FC = () => {
  return (
    <ThemeConsumer>
      {(theme) => <ClipLoader color={theme.actionloader} size={20} />}
    </ThemeConsumer>
  );
};
