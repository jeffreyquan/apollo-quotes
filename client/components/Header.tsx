import Link from "next/link";
import { useState, useEffect } from "react";
import styled from "styled-components";
import { IoIosRocket } from "react-icons/io";
import { Navbar } from "./Navbar";

const Logo = styled.div`
  font-size: 3rem;
  margin-right: auto;
  a {
    display: flex;
    align-items: center;
  }
`;

const StyledHeader = styled.header`
  position: sticky;
  background-color: ${(props) =>
    props.sticky ? props.theme.white : "transparent"};
  box-shadow: ${(props) =>
    props.sticky ? `0 0.2rem 0.5rem rgba(0,0,0,0.5)` : ""};
  top: 0;
  display: flex;
  align-items: center;
  transition: all 0.3s ease-in-out;
  line-height: ${(props) => (props.sticky ? "3rem" : "5rem")};
  max-height: 9rem;
`;

export const Header = () => {
  const [sticky, setSticky] = useState(false);

  useEffect(() => {
    window.addEventListener("scroll", () => {
      if (window.scrollY > 0) {
        setSticky(true);
      } else {
        setSticky(false);
      }
    });
  }, []);

  return (
    <StyledHeader sticky={sticky}>
      <Logo>
        <Link href="/">
          <a>
            <IoIosRocket />
            Apollo Quotes
          </a>
        </Link>
      </Logo>
      <Navbar />
    </StyledHeader>
  );
};
