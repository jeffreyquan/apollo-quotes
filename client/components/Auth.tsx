import { createContext, useState, useEffect } from "react";
import { useQuery, gql } from "@apollo/client";

const CURRENT_USER_QUERY = gql`
  query {
    userProfile {
      id
      name
      username
      email
      role
    }
  }
`;

export const AuthContext = createContext({
  checkingAuth: true,
  user: null,
  setUser: null,
});

export const AuthProvider = ({ children }) => {
  const [checkingAuth, setCheckingAuth] = useState(true);
  const [user, setUser] = useState(null);

  const { data, loading, error } = useQuery(CURRENT_USER_QUERY);

  useEffect(() => {
    if (typeof data !== undefined) {
      setCheckingAuth(false);
      if (data) {
        setUser(data.userProfile);
      }
    }
  }, [data]);

  const value = { checkingAuth, user, setUser };

  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
};
