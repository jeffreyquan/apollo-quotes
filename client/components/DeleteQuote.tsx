import React from "react";
import { gql, useMutation } from "@apollo/client";
import { BsExclamationCircle } from "react-icons/bs";
import { useRouter } from "next/router";
import { ActionLoader } from "./ActionLoader";

const DELETE_QUOTE_MUTATION = gql`
  mutation DELETE_QUOTE_MUTATION($id: ID!) {
    deleteQuote(id: $id) {
      id
    }
  }
`;

interface DeleteQuoteProps {
  id: string;
}

export const DeleteQuote: React.FC<DeleteQuoteProps> = ({ id, children }) => {
  const [deleteQuote, { error, loading }] = useMutation(DELETE_QUOTE_MUTATION, {
    variables: { id },
    update(cache, { data: { deleteQuote } }) {
      cache.modify({
        fields: {
          quotes(
            existing = {
              __typename: "QuotesConnection",
              totalCount: 0,
              pageInfo: {
                endCursor: null,
                hasMore: true,
              },
              quotes: [],
            },
            { readField }
          ) {
            const updatedQuotes = existing.quotes.filter(
              (quoteRef) => readField("id", quoteRef) !== deleteQuote.id
            );
            return {
              ...existing,
              quotes: [...updatedQuotes],
            };
          },
        },
      });
    },
  });

  if (loading) return <ActionLoader />;
  if (error) return <BsExclamationCircle />;

  const router = useRouter();

  const handleClick = async (e) => {
    e.preventDefault();
    try {
      await deleteQuote();
      router.push({
        pathname: "/quotes",
        query: { delete: "success" },
      });
    } catch (err) {
      return <BsExclamationCircle />;
    }
  };

  return (
    <div
      data-testid="deleteButton"
      onClick={(e) => {
        if (confirm("Please confirm you want to delete this quote.")) {
          handleClick(e);
        }
      }}
    >
      {children}
    </div>
  );
};
