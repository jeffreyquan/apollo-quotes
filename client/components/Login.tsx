import { useState, useEffect, useContext } from "react";
import { useRouter } from "next/router";
import { gql, useMutation } from "@apollo/client";
import { AuthContext } from "./Auth";
import { useForm } from "../lib/useForm";
import { Form } from "../styles/Form";
import { FormContainer } from "../styles/FormContainer";
import { FormTitle } from "../styles/FormTitle";

const LOGIN_MUTATION = gql`
  mutation LOGIN_MUTATION($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      id
      name
      email
      username
    }
  }
`;

export const Login = () => {
  const { inputs, handleChange, resetForm } = useForm({
    email: "",
    password: "",
  });

  const { email, password } = inputs;

  const [loadingPage, setLoadingPage] = useState(true);

  let { user, setUser } = useContext(AuthContext);

  const router = useRouter();

  useEffect(() => {
    if (user) {
      if (router.query.redirect) {
        router.back();
      } else {
        router.push("/");
      }
    } else {
      setLoadingPage(false);
    }
  }, [user]);

  const [login, { error, loading }] = useMutation(LOGIN_MUTATION, {
    variables: inputs,
  });

  // TODO: create error component to display any login errors
  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const res = await login();
      setUser(res.data.login);
    } catch (err) {
      console.log(err);
    }
  };

  return !loadingPage ? (
    <FormContainer>
      <Form onSubmit={handleSubmit}>
        <FormTitle>Login</FormTitle>
        <fieldset disabled={loading} aria-busy={loading}>
          <label htmlFor="email">
            Email
            <input
              type="email"
              name="email"
              placeholder="Email"
              value={email}
              onChange={handleChange}
              autoComplete="email"
            />
          </label>
          <label htmlFor="password">
            Password
            <input
              type="password"
              name="password"
              placeholder="Password"
              value={password}
              onChange={handleChange}
              autoComplete="current-password"
            />
          </label>
          <input type="submit" value="Sign In" />
        </fieldset>
      </Form>
    </FormContainer>
  ) : (
    <div>Loading....</div>
  );
};
