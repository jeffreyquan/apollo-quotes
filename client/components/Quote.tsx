import { useContext, useState } from "react";
import Link from "next/link";
import { useRouter } from "next/router";
import { gql, useMutation } from "@apollo/client";
import styled from "styled-components";
import { BsHeart, BsHeartFill } from "react-icons/bs";
import { FiDelete, FiEdit } from "react-icons/fi";
import { QuoteStyles } from "../styles/QuoteStyles";
import { QuoteTag } from "../styles/QuoteTag";
import { Quote as QuoteType } from "../types";
import { AuthContext } from "./Auth";
import { DeleteQuote } from "./DeleteQuote";

export const QuoteBody = styled.div`
  display: grid;
  grid-template-columns: repeat(8, 1fr);
  grid-gap: 1rem;
`;

export const QuoteImg = styled.div`
  grid-column: 1 / 4;
`;

export const QuoteContent = styled.div`
  grid-column: 4 / 9;
`;

export const QuoteFooter = styled.div`
  display: flex;
  align-items: center;
  margin-right: auto;
`;

export const Controls = styled.div`
  display: flex;
  align-items: center;

  & > div {
    display: flex;
    align-items: center;
    font-size: 1.8rem;
    svg {
      margin-left: 1.2rem;
    }
  }
`;

const TagList = styled.div`
  display: inline-block;
  margin-right: auto;
`;

export const LIKE_MUTATION = gql`
  mutation LIKE_MUTATION($quoteId: ID!) {
    likeQuote(quoteId: $quoteId) {
      id
      quote {
        id
        likes {
          id
          user {
            id
            username
          }
        }
      }
      user {
        id
        username
      }
    }
  }
`;

export const Quote = ({ quote }) => {
  const { id, author, content, image, tags, likes, submittedBy, slug } = quote;

  const [like] = useMutation(LIKE_MUTATION, {
    variables: {
      quoteId: id,
    },
    update(cache, { data: { likeQuote } }) {
      const likedQuote: QuoteType = cache.readFragment({
        id: `Quote:${id}`,
        fragment: gql`
          fragment Like on Quote {
            likes {
              id
            }
          }
        `,
      });
      const updatedLikes = likedQuote.likes;
      const foundLike = updatedLikes.some((like) => like.id === likeQuote.id);
      if (!foundLike) {
        cache.evict({
          id: `Like:${likeQuote.id}`,
        });
      }
    },
  });

  const [loading, setLoading] = useState(true);

  let { user } = useContext(AuthContext);

  let liked;

  if (user && likes) {
    liked = likes.some((like) => like.user.id === user.id);
  }

  let submitted;

  if (user) {
    submitted = submittedBy.id === user.id ? true : false;
  }

  const router = useRouter();

  const fetchQuoteWithTag = (
    e: React.MouseEvent<HTMLDivElement>,
    tagName: string
  ) => {
    e.preventDefault();
    router.push({
      pathname: "/quotes",
      query: { tag: tagName },
    });
  };

  const likeQuote = async (e) => {
    e.preventDefault();
    try {
      await like();
    } catch (e) {
      console.log(e);
    }
  };

  const EditLink = ({ children, href, as, ...props }) => (
    <Link href={href} as={as}>
      <div {...props}>{children}</div>
    </Link>
  );

  return (
    <QuoteStyles loading={loading ? "true" : undefined}>
      <QuoteBody>
        {image && (
          <QuoteImg>
            <img src={image} alt={author} onLoad={() => setLoading(false)} />
          </QuoteImg>
        )}
        <QuoteContent>
          <p>
            &#8220;<span>{content}</span>&#8221;
          </p>
          <p>
            &mdash; <span>{author}</span>
          </p>
        </QuoteContent>
      </QuoteBody>
      <QuoteFooter>
        <TagList>
          {tags.map((tag) => (
            <QuoteTag
              key={tag.id}
              onClick={(e) => fetchQuoteWithTag(e, tag.name)}
            >
              {tag.name}
            </QuoteTag>
          ))}
        </TagList>
        <Controls>
          <div>
            <span>{likes.length > 0 && likes.length}</span>
            {liked ? (
              <BsHeartFill data-test="like-btn" onClick={(e) => likeQuote(e)} />
            ) : (
              <BsHeart data-test="like-btn" onClick={(e) => likeQuote(e)} />
            )}
          </div>
          {submitted && (
            <>
              <EditLink href="/quotes/[...slug]" as={`/quotes/${slug}/edit`}>
                <FiEdit />
              </EditLink>

              <DeleteQuote id={id}>
                <FiDelete />
              </DeleteQuote>
            </>
          )}
        </Controls>
      </QuoteFooter>
    </QuoteStyles>
  );
};
