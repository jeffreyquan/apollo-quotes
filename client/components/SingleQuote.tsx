import { gql, useQuery } from "@apollo/client";
import styled from "styled-components";
import { PageLoader } from "./PageLoader";
import { Quote } from "../components/Quote";

export const SINGLE_QUOTE_QUERY = gql`
  query SINGLE_QUOTE_QUERY($slug: String!) {
    quote(slug: $slug) {
      id
      author
      content
      image
      largeImage
      submittedBy {
        id
      }
      tags {
        id
        name
      }
      likes {
        id
        user {
          id
          username
        }
      }
      slug
    }
  }
`;

const SingleQuoteStyles = styled.div`
  display: flex;
  align-items: center;
  height: 100%;
  max-width: ${(props) => props.theme.maxWidth};
  margin: 0 auto;
`;

interface SingleQuoteProps {
  slug: string;
}

export const SingleQuote: React.FC<SingleQuoteProps> = ({ slug }) => {
  const { data, loading, error } = useQuery(SINGLE_QUOTE_QUERY, {
    variables: {
      slug,
    },
  });

  if (loading) return <PageLoader />;
  if (error) return <div>Error...</div>;

  const updatedQuote = { ...data.quote };
  updatedQuote.image = updatedQuote.largeImage;

  return (
    <SingleQuoteStyles>
      <Quote quote={updatedQuote} />
    </SingleQuoteStyles>
  );
};
