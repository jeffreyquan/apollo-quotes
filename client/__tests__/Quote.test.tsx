import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MockedProvider } from "@apollo/client/testing";
import { useRouter } from "next/router";
import { Quote, LIKE_MUTATION } from "../components/Quote";
import { SINGLE_QUOTE_QUERY } from "../components/SingleQuote";
import { AuthProvider } from "../components/Auth";
import { CURRENT_USER_QUERY } from "../components/Auth";
import {
  fakeCache,
  fakeQuoteWithoutUserLike,
  fakeQuoteWithUserLike,
} from "../lib/testUtils";

const mockRouterPush = jest.fn();

const quoteWithoutUserLike = fakeQuoteWithoutUserLike();

const quoteWithUserLike = fakeQuoteWithUserLike();

jest.mock("next/router", () => ({
  ...jest.requireActual("next/router"),
  useRouter: () => ({
    push: mockRouterPush,
  }),
}));

describe("Quote", () => {
  it("renders and matches snapshot when user is not signed in", () => {
    const { container } = render(
      <MockedProvider>
        <Quote quote={quoteWithoutUserLike} />
      </MockedProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders and matches snapshot when user is signed in and quote belongs to user", async () => {
    let currentUserQueryCalled = false;
    const mocks = [
      {
        request: { query: CURRENT_USER_QUERY },
        result: () => {
          currentUserQueryCalled = true;

          return {
            data: {
              userProfile: {
                id: "abc123",
                name: "Jeffrey",
                username: "jeffrey",
                email: "jeffrey@example.com",
                role: "ADMIN",
                __typename: "User",
              },
            },
          };
        },
      },
    ];

    const { container, getByTestId } = render(
      <MockedProvider mocks={mocks}>
        <AuthProvider>
          <Quote quote={quoteWithoutUserLike} />
        </AuthProvider>
      </MockedProvider>
    );

    await waitFor(() => {
      expect(currentUserQueryCalled).toBe(true);
    });

    expect(getByTestId("editButton")).toBeInTheDocument();

    expect(getByTestId("deleteButton")).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("renders the image", () => {
    render(
      <MockedProvider>
        <Quote quote={quoteWithoutUserLike} />
      </MockedProvider>
    );
    const img = screen.getByAltText(quoteWithoutUserLike.author);
    expect(img).toBeInTheDocument();
  });

  it("renders the content", () => {
    const { getByTestId } = render(
      <MockedProvider>
        <Quote quote={quoteWithoutUserLike} />
      </MockedProvider>
    );

    expect(getByTestId("quoteContent")).toHaveTextContent(
      quoteWithoutUserLike.content
    );
  });

  it("renders the author", () => {
    const { getByTestId } = render(
      <MockedProvider>
        <Quote quote={quoteWithoutUserLike} />
      </MockedProvider>
    );

    expect(getByTestId("quoteAuthor")).toHaveTextContent(
      quoteWithoutUserLike.author
    );
  });

  it("renders the right number of tags", () => {
    const { getAllByTestId } = render(
      <MockedProvider>
        <Quote quote={quoteWithoutUserLike} />
      </MockedProvider>
    );

    expect(getAllByTestId("quoteTag")).toHaveLength(2);
  });

  it("renders the tags properly", () => {
    const { getAllByTestId } = render(
      <MockedProvider>
        <Quote quote={quoteWithoutUserLike} />
      </MockedProvider>
    );
    for (let i = 0; i < quoteWithoutUserLike.tags.length; i++) {
      const quoteTags = getAllByTestId("quoteTag");
      expect(quoteTags[i]).toHaveTextContent(quoteWithoutUserLike.tags[i].name);
    }
  });

  it("renders the correct number of likes", () => {
    const { getByTestId } = render(
      <MockedProvider>
        <Quote quote={quoteWithoutUserLike} />
      </MockedProvider>
    );

    expect(getByTestId("likeCount")).toHaveTextContent(
      `${quoteWithoutUserLike.likes.length}`
    );
  });

  it("increases the like count by 1 when quote is liked", async () => {
    const cache = fakeCache();

    cache.writeQuery({
      query: SINGLE_QUOTE_QUERY,
      variables: { slug: "forrest-gump" },
      data: {
        quote: {
          ...quoteWithoutUserLike,
        },
      },
    });

    let currentUserQueryCalled = false;
    let likeMutationCalled = false;

    const userLikedQuoteMocks = [
      {
        request: { query: CURRENT_USER_QUERY },
        result: () => {
          currentUserQueryCalled = true;

          return {
            data: {
              userProfile: {
                id: "abc123",
                name: "Jeffrey",
                username: "jeffrey",
                email: "jeffrey@example.com",
                role: "ADMIN",
                __typename: "User",
              },
            },
          };
        },
      },
      {
        request: {
          query: LIKE_MUTATION,
          variables: {
            quoteId: quoteWithoutUserLike.id,
          },
        },
        result: () => {
          likeMutationCalled = true;

          return {
            data: {
              likeQuote: {
                id: "like123",
                quote: {
                  id: quoteWithoutUserLike.id,
                  likes: [
                    {
                      id: "like123",
                      user: {
                        id: "abc123",
                        username: "jeffrey",
                        __typename: "User",
                      },
                      __typename: "Like",
                    },
                    ...quoteWithoutUserLike.likes,
                  ],
                  __typename: "Quote",
                },
                user: {
                  id: "abc123",
                  username: "jeffrey",
                  __typename: "User",
                },
                __typename: "Like",
              },
            },
          };
        },
      },
    ];

    const { getByTestId, getByText, rerender } = render(
      <MockedProvider mocks={userLikedQuoteMocks} cache={cache}>
        <AuthProvider>
          <Quote quote={quoteWithoutUserLike} />
        </AuthProvider>
      </MockedProvider>
    );

    await waitFor(() => {
      expect(currentUserQueryCalled).toBe(true);
    });

    expect(getByTestId("likeCount")).toHaveTextContent(
      `${quoteWithoutUserLike.likes.length}`
    );

    const likeButton = getByTestId("likeButton");

    await userEvent.click(likeButton);

    await waitFor(() => {
      expect(likeMutationCalled).toBe(true);
    });

    const { quote } = cache.readQuery({
      query: SINGLE_QUOTE_QUERY,
      variables: { slug: "forrest-gump" },
    });

    rerender(
      <MockedProvider mocks={userLikedQuoteMocks} cache={cache}>
        <AuthProvider>
          <Quote quote={quote} />
        </AuthProvider>
      </MockedProvider>
    );

    await waitFor(() => {
      expect(
        getByText(`${quoteWithoutUserLike.likes.length + 1}`)
      ).toBeInTheDocument();
    });
  });

  it("decreases the like count by 1 when quote is unliked", async () => {
    const cache = fakeCache();

    cache.writeQuery({
      query: SINGLE_QUOTE_QUERY,
      variables: { slug: "forrest-gump" },
      data: {
        quote: {
          ...quoteWithUserLike,
        },
      },
    });

    let currentUserQueryCalled = false;
    let likeMutationCalled = false;

    const userUnlikedQuoteMocks = [
      {
        request: { query: CURRENT_USER_QUERY },
        result: () => {
          currentUserQueryCalled = true;

          return {
            data: {
              userProfile: {
                id: "abc123",
                name: "Jeffrey",
                username: "jeffrey",
                email: "jeffrey@example.com",
                role: "ADMIN",
                __typename: "User",
              },
            },
          };
        },
      },
      {
        request: {
          query: LIKE_MUTATION,
          variables: {
            quoteId: quoteWithUserLike.id,
          },
        },
        result: () => {
          likeMutationCalled = true;

          return {
            data: {
              likeQuote: {
                id: "like123",
                quote: {
                  id: quoteWithUserLike.id,
                  likes: [...quoteWithoutUserLike.likes], // result of unliking a quote is that the quote will no longer have a like belonging to the user
                  __typename: "Quote",
                },
                user: {
                  id: "abc123",
                  username: "jeffrey",
                  __typename: "User",
                },
                __typename: "Like",
              },
            },
          };
        },
      },
    ];

    const { getByTestId, getByText, rerender } = render(
      <MockedProvider mocks={userUnlikedQuoteMocks} cache={cache}>
        <AuthProvider>
          <Quote quote={quoteWithUserLike} />
        </AuthProvider>
      </MockedProvider>
    );

    await waitFor(() => {
      expect(currentUserQueryCalled).toBe(true);
    });

    expect(getByTestId("likeCount")).toHaveTextContent(
      `${quoteWithUserLike.likes.length}`
    );

    const unlikeButton = screen.queryByTestId("unlikeButton");

    expect(unlikeButton).toBeInTheDocument();

    await userEvent.click(unlikeButton);

    await waitFor(() => {
      expect(likeMutationCalled).toBe(true);
    });

    const { quote } = cache.readQuery({
      query: SINGLE_QUOTE_QUERY,
      variables: { slug: "forrest-gump" },
    });

    rerender(
      <MockedProvider mocks={userUnlikedQuoteMocks} cache={cache}>
        <AuthProvider>
          <Quote quote={quote} />
        </AuthProvider>
      </MockedProvider>
    );

    await waitFor(() => {
      expect(
        getByText(`${quoteWithUserLike.likes.length - 1}`)
      ).toBeInTheDocument();
    });
  });

  it("navigates to correct route when tags are clicked", async () => {
    const { getByText } = render(
      <MockedProvider>
        <Quote quote={quoteWithoutUserLike} />
      </MockedProvider>
    );

    for (let i = 0; i < quoteWithoutUserLike.tags.length; i++) {
      const tag = getByText(quoteWithoutUserLike.tags[i].name);

      const router = useRouter();

      await userEvent.click(tag);

      await waitFor(() =>
        expect(router.push).toHaveBeenCalledWith({
          pathname: "/quotes",
          query: { tag: quoteWithoutUserLike.tags[i].name },
        })
      );
    }
  });
});
