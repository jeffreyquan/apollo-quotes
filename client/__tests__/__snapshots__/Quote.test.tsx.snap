// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote renders and matches snapshot when user is not signed in 1`] = `
<div>
  <div
    class="QuoteStyles-sc-17p689u-0 cEqKCH"
    loading="true"
  >
    <div
      class="Quote__QuoteBody-sc-1h9v3ld-0 cnXFty"
    >
      <div
        class="Quote__QuoteImg-sc-1h9v3ld-1 llufdZ"
      >
        <img
          alt="Forrest Gump, Forrest Gump"
          src="https://www.fillmurray.com/200/300"
        />
      </div>
      <div
        class="Quote__QuoteContent-sc-1h9v3ld-2 jqRyuo"
      >
        <p>
          “
          <span
            data-testid="quoteContent"
          >
            Life was like a box of chocolates. You never know what you're gonna get.
          </span>
          ”
        </p>
        <p>
          — 
          <span
            data-testid="quoteAuthor"
          >
            Forrest Gump, Forrest Gump
          </span>
        </p>
      </div>
    </div>
    <div
      class="Quote__QuoteFooter-sc-1h9v3ld-3 iQBsdY"
    >
      <div
        class="Quote__TagList-sc-1h9v3ld-5 drKxMP"
      >
        <div
          class="QuoteTag-sc-4kw1b-0 kwroDf"
          data-testid="quoteTag"
        >
          movie
        </div>
        <div
          class="QuoteTag-sc-4kw1b-0 kwroDf"
          data-testid="quoteTag"
        >
          life
        </div>
      </div>
      <div
        class="Quote__Controls-sc-1h9v3ld-4 wBSRR"
      >
        <div>
          <span
            data-testid="likeCount"
          >
            2
          </span>
          <svg
            data-testid="likeButton"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8 2.748l-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 01.176-.17C12.72-3.042 23.333 4.867 8 15z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Quote renders and matches snapshot when user is signed in and quote belongs to user 1`] = `
<div>
  <div
    class="QuoteStyles-sc-17p689u-0 cEqKCH"
    loading="true"
  >
    <div
      class="Quote__QuoteBody-sc-1h9v3ld-0 cnXFty"
    >
      <div
        class="Quote__QuoteImg-sc-1h9v3ld-1 llufdZ"
      >
        <img
          alt="Forrest Gump, Forrest Gump"
          src="https://www.fillmurray.com/200/300"
        />
      </div>
      <div
        class="Quote__QuoteContent-sc-1h9v3ld-2 jqRyuo"
      >
        <p>
          “
          <span
            data-testid="quoteContent"
          >
            Life was like a box of chocolates. You never know what you're gonna get.
          </span>
          ”
        </p>
        <p>
          — 
          <span
            data-testid="quoteAuthor"
          >
            Forrest Gump, Forrest Gump
          </span>
        </p>
      </div>
    </div>
    <div
      class="Quote__QuoteFooter-sc-1h9v3ld-3 iQBsdY"
    >
      <div
        class="Quote__TagList-sc-1h9v3ld-5 drKxMP"
      >
        <div
          class="QuoteTag-sc-4kw1b-0 kwroDf"
          data-testid="quoteTag"
        >
          movie
        </div>
        <div
          class="QuoteTag-sc-4kw1b-0 kwroDf"
          data-testid="quoteTag"
        >
          life
        </div>
      </div>
      <div
        class="Quote__Controls-sc-1h9v3ld-4 wBSRR"
      >
        <div>
          <span
            data-testid="likeCount"
          >
            2
          </span>
          <svg
            data-testid="likeButton"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8 2.748l-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 01.176-.17C12.72-3.042 23.333 4.867 8 15z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          data-testid="editButton"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
            />
            <path
              d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
            />
          </svg>
        </div>
        <div
          data-testid="deleteButton"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 4H8l-7 8 7 8h13a2 2 0 0 0 2-2V6a2 2 0 0 0-2-2z"
            />
            <line
              x1="18"
              x2="12"
              y1="9"
              y2="15"
            />
            <line
              x1="12"
              x2="18"
              y1="9"
              y2="15"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
